tests/providers/docker/decorators/test_docker.py

```diff
@@ -60,6 +60,21 @@ def f(num_results):
         assert isinstance(result, list)
         assert len(result) == 50
 
+    def test_basic_docker_operator_with_template_fields(
+        , dag_maker):
+        @task.docker(image="python:3.9-slim", container_name='python_{{dag_run.dag_id}}', auto_remove="force")
+        def f():
+            raise RuntimeError("Should not executed") 
+
+        with dag_maker():
+            ret = f()
+
+        dr = dag_maker.create_dagrun()
+        ti = dr.get_task_instances()[0]
+        rendered_op_kwargs = ti.render_templates().op_kwargs
+        assert rendered_op_kwargs["container_name"] == f"python_{dr.dag_id}"
```

Comment:
Ok, not last. I tested your branch locally and it failed for different reasons

```suggestion
    def test_basic_docker_operator_with_template_fields(self, dag_maker):
        @task.docker(image="python:3.9-slim", container_name='python_{{dag_run.dag_id}}')
        def f():
            raise RuntimeError("Should not executed") 

        with dag_maker():
            ret = f()

        dr = dag_maker.create_dagrun()
        ti = TaskInstance(task=ret.operator, run_id=dr.run_id)
        rendered = ti.render_templates()
        assert rendered.container_name == f"python_{dr.dag_id}"
```

Please note, you also need to import `from airflow.models import TaskInstance`

Reply:
Thanks for the help Taragolis, I've learnt a lot from you